code/modules/mob/living/simple_animal/hostile/mining/elites/herald.dm

```diff
@@ -0,0 +1,326 @@
+#define HERALD_TRISHOT 1
+#define HERALD_DIRECTIONALSHOT 2
+#define HERALD_TELESHOT 3
+#define HERALD_MIRROR 4
+
+/**
+ * # Herald
+ *
+ * A slow-moving projectile user with a few tricks up it's sleeve.  Less unga-bunga than Colossus, with more cleverness in it's fighting style.
+ * As it's health gets lower, the amount of projectiles fired per-attack increases.
+ * It's attacks are as follows:
+ * - Fires three projectiles in a given direction.
+ * - Fires a spread in every cardinal and diagonal direction at once, then does it again after a bit.
+ * - Shoots a single, golden bolt.  Wherever it lands, the herald will be teleported to the location.
+ * - Spawns a mirror which reflects projectiles directly at the target.
+ * Herald is a more concentrated variation of the Colossus fight, having less projectiles overall, but more focused attacks.
+ */
+
+/mob/living/simple_animal/hostile/asteroid/elite/herald
+	name = "herald"
+	desc = "A monstrous beast which fires deadly projectiles at threats and prey."
+	icon_state = "herald"
+	icon_living = "herald"
+	icon_aggro = "herald"
+	icon_dead = "herald_dying"
+	icon_gib = "syndicate_gib"
+	maxHealth = 1000
+	health = 1000
+	melee_damage_lower = 20
+	melee_damage_upper = 20
+	attacktext = "preaches to"
+	attack_sound = 'sound/magic/ratvar_attack.ogg'
+	throw_message = "doesn't affect the purity of"
+	speed = 2
+	move_to_delay = 10
+	universal_speak = TRUE
+	mouse_opacity = MOUSE_OPACITY_ICON
+	death_sound = 'sound/misc/demon_dies.ogg'
+	deathmessage = "begins to shudder as it becomes transparent..."
+	loot_drop = /obj/item/clothing/accessory/necklace/herald_cloak
+
+	can_talk = 1
+
+	attack_action_types = list(/datum/action/innate/elite_attack/herald_trishot,
+								/datum/action/innate/elite_attack/herald_directionalshot,
+								/datum/action/innate/elite_attack/herald_teleshot,
+								/datum/action/innate/elite_attack/herald_mirror)
+
+	var/mob/living/simple_animal/hostile/asteroid/elite/herald/mirror/my_mirror = null
+	var/is_mirror = FALSE
+
+/mob/living/simple_animal/hostile/asteroid/elite/herald/death()
+	. = ..()
+	if(!is_mirror)
+		addtimer(CALLBACK(src, .proc/become_ghost), 0.8 SECONDS)
+	if(!isnull(my_mirror))
+		qdel(my_mirror)
```

Comment:
should be in a destroy proc. this should also be QDEL_NULL().
and the mirror should null out its master on destroy

Reply:
done!